assets/src/modules/Snapping.js

```diff
@@ -0,0 +1,117 @@
+import { mainLizmap, mainEventDispatcher } from '../modules/Globals.js';
+
+export default class Snapping {
+
+    constructor() {
+
+        this._maxFeatures = 1000;
+        this._restrictToMapExtent = true;
+        this._config = undefined;
+
+        // Create layer to store snap features
+        const snapLayer = new OpenLayers.Layer.Vector('snaplayer', {
+            styleMap: new OpenLayers.StyleMap({
+                pointRadius: 2,
+                fill: false,
+                stroke: true,
+                strokeWidth: 3,
+                strokeColor: 'red',
+                strokeOpacity: 0.8
+            })
+        });
+
+        mainLizmap.lizmap3.map.addLayer(snapLayer);
+
+        const snapControl = new OpenLayers.Control.Snapping({
+            layer: mainLizmap.edition.editLayer,
+            targets: [{
+                layer: snapLayer
+            }]
+        });
+        mainLizmap.lizmap3.map.addControls([snapControl]);
+        mainLizmap.lizmap3.controls['snapControl'] = snapControl;
+
+        // Activate snap when a layer is edited
+        mainEventDispatcher.addListener(
+            () => {
+                // Get snapping configuration for edited layer
+                for (const editionLayer in mainLizmap.config.editionLayers) {
+                    if (mainLizmap.config.editionLayers.hasOwnProperty(editionLayer)) {
+                        if (mainLizmap.config.editionLayers[editionLayer].layerId === mainLizmap.edition.layerId){
+                            const editionLayerConfig = mainLizmap.config.editionLayers[editionLayer];
+                            if (editionLayerConfig.hasOwnProperty('snap_layers')){
+                                this._config = {
+                                    'snap_layers': editionLayerConfig.snap_layers.split(','),
```

Comment:
I will update to a list. Can you support the list? It would be better.

Reply:
Yes I will.

Reply:
Done https://github.com/3liz/lizmap-plugin/pull/252/commits/d21bc55bcb9720cd51af7a517cb1a6be1650929b

Reply:
Done